### PR TITLE
Merge ColorSensorSubsystem into master

### LINKED ISSUE #2: Colour Sensor
Using FRC provided colour sensor, write software to access the sensor

### PR DESCRIPTION
**Pull request by nrgill28**
Added a subsystem for interacting with the color sensor and color matching classes.

Closes #2 .

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -10,41 +10,41 @@
...
+ * This class is where the bulk of the robot should be declared. Since
+ * Command-based is a "declarative" paradigm, very little robot logic should
+ * actually be handled in the {@link Robot} periodic methods (other than the
+ * scheduler calls). Instead, the structure of the robot (including subsystems,
+ * commands, and button mappings) should be declared here.
  */
 public class RobotContainer {
   // The robot's subsystems and commands are defined here...
   private final ExampleSubsystem m_exampleSubsystem = new ExampleSubsystem();
+  private final ColorSensorSubsystem m_ColorSensorSubsystem = new ColorSensorSubsystem();
```
Comment (Zinka010) [inline src/main/java/frc/robot/RobotContainer.java]: Necessary if unused?

File: src/main/java/frc/robot/subsystems/ColorSensorSubsystem.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+  /**
+   * Copied values from RevRobotic's ColorMatch example
+   * https://github.com/REVrobotics/Color-Sensor-v3-Examples/blob/master/Java/Color%20Match/src/main/java/frc/robot/Robot.java
+   */
+  private final Color kBlueTarget = ColorMatch.makeColor(0.143, 0.427, 0.429);
+  private final Color kGreenTarget = ColorMatch.makeColor(0.197, 0.561, 0.240);
+  private final Color kRedTarget = ColorMatch.makeColor(0.561, 0.232, 0.114);
+  private final Color kYellowTarget = ColorMatch.makeColor(0.361, 0.524, 0.113);
+  private final Color kWhiteTarget = ColorMatch.makeColor(0.26, 0.25, 0.48);
```
Comment (Zinka010) [inline src/main/java/frc/robot/subsystems/ColorSensorSubsystem.java]: Remove this if it doesn't actually read the white value correctly

**Review by Zinka010 — CHANGES_REQUESTED** [review]

**Review by Lockscout — APPROVED** [review]
Looks good

**Review by Zinka010 — APPROVED** [review]
Nice

**Pull request merged**